### PR TITLE
[docs] Update Java 17 compatibility docs

### PR DESCRIPTION
**Pull request by mbalassi**
Updates docs to match current state. In practice we already see usage with Java 17.

### CODE REVIEW

File: docs/content/docs/deployment/java_compatibility.md [hunk 1]
```diff
@@ -50,7 +50,8 @@ The following Flink features have not been tested with Java 11:
 
 ## Java 17
 
-Experimental support for Java 17 was added in 1.18.0. ([FLINK-15736](https://issues.apache.org/jira/browse/FLINK-15736))
+Since version 1.19.0 Flink works reasonably well with Java 17, with the caveat that the integration is incomplete.
```
Comment (rmetzger) [inline docs/content/docs/deployment/java_compatibility.md:53]: I would keep the "Experimental support for Java 17 was added in 1.18.0. " line, as it gives more context.

if you just say "works reasonable well", it is not very convincing, and rather sounds like we don't know what we are doing here or what the status is.
Do we know what the caveats are?

File: docs/content/docs/deployment/java_compatibility.md [hunk 2]
```diff
@@ -50,7 +50,8 @@ The following Flink features have not been tested with Java 11:
...
+Importantly, as per popular demand Java Records are supported ([FLINK-32380](https://issues.apache.org/jira/browse/FLINK-32380)).
```
Comment (rmetzger) [inline docs/content/docs/deployment/java_compatibility.md:54]:
```suggestion
In Flink 1.19, we added support for Java Records ([FLINK-32380](https://issues.apache.org/jira/browse/FLINK-32380)).
```

**Comment by rmetzger** [comment]
Thanks. Overall, I think the tone is a bit too much like a blog post, and a not enough like a documentation that you use as a reference.

File: docs/content/docs/deployment/java_compatibility.md (hunk 1 above)
Comment (mbalassi) [reply docs/content/docs/deployment/java_compatibility.md:53]: Fair enough. Caveats are listed in the `Known issues` section.

**Comment by mbalassi** [comment]
> Thanks. Overall, I think the tone is a bit too much like a blog post, and a not enough like a documentation that you use as a reference.

Thanks, updated accordingly.

**Review by rmetzger — APPROVED** [review]

**Pull request merged**